core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -421,10 +424,32 @@ public void start(boolean isLeader) throws IOException {
       mLostFilesDetectionService = getExecutorService().submit(new HeartbeatThread(
           HeartbeatContext.MASTER_LOST_FILES_DETECTION, new LostFilesDetectionHeartbeatExecutor(),
           Configuration.getInt(PropertyKey.MASTER_HEARTBEAT_INTERVAL_MS)));
+      mStartupConsistencyCheck = getExecutorService().submit(new Callable<List<AlluxioURI>>() {
+        @Override
+        public List<AlluxioURI> call() throws Exception {
+          return checkConsistency(new AlluxioURI("/"), CheckConsistencyOptions.defaults());
+        }
+      });
     }
   }
 
   /**
+   * @return The list of inconsistent files at start up, null if the check has not completed or
+   *         failed.
+   */
+  public List<AlluxioURI> getStartupConsistencyCheck() {
+    boolean isDone = mStartupConsistencyCheck.isDone();
+    if (isDone) {
+      try {
+        return mStartupConsistencyCheck.get();
+      } catch (Exception e) {
+        // fall through
```

Comment:
Good point, I've added a log message. I've also made a follow up task to expose this info to the user.